src/evaluators/UnaryExpression.js

```diff
@@ -109,23 +113,25 @@ function generateRuntimeCall(ref: Reference | Value, ast: BabelNodeUnaryExpressi
   if (Environment.IsPropertyReference(realm, ref)) {
     value = Environment.GetBase(realm, ref);
     invariant(value instanceof Value);
-    if (!value.isSimpleObject()) {
+    if (value.mightBeObject() && !value.isSimpleObject()) {
       Leak.leakValue(realm, value, ast.loc);
     }
     propertyName = Environment.GetReferencedName(realm, ref);
   } else {
     value = Environment.GetValue(realm, ref);
-    invariant(value instanceof Value);
     Leak.leakValue(realm, value, ast.loc);
   }
 
   return AbstractValue.createTemporalFromBuildFunction(realm, Value, [value], nodes => {
     let arg;
     if (propertyName) {
-      invariant(typeof propertyName === "string");
-      arg = t.isValidIdentifier(propertyName)
-        ? t.memberExpression(nodes[0], t.identifier(propertyName), false)
-        : t.memberExpression(nodes[0], t.stringLiteral(propertyName), true);
+      if (typeof propertyName === "string") {
+        arg = t.isValidIdentifier(propertyName)
+          ? t.memberExpression(nodes[0], t.identifier(propertyName), false)
+          : t.memberExpression(nodes[0], t.stringLiteral(propertyName), true);
+      } else {
+        arg = t.memberExpression(nodes[0], nodes[1], true);
```

Comment:
nodes[1] won't be defined unless you pass propertyName as an argument in line 125. See generateRuntime call in CallExpression.js.